Tasks/AzureTestPlanV0/ciEventLogger.ts

```diff
@@ -0,0 +1,33 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+
+const area: string = 'TestManagement';
+const feature: string = 'AzureTestPlanTask';
+
+function getDefaultProps() {
+    return {
+        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        releaseid: tl.getVariable('Release.ReleaseId'),
+        builduri: tl.getVariable('Build.BuildUri'),
+        buildid: tl.getVariable('Build.Buildid')
+    };
+}
+
+export function publishEvent(properties: { [key: string]: any }): void {
+    try {
+        tl.assertAgent('2.144.0');
+        publishTelemetry(area, feature, Object.assign(getDefaultProps(), properties));
+
+    } catch (err) {
+        tl.debug('Unable to publish telemetry due to lower agent version.');
```

Comment:
will the error message be same in all the cases?

Reply:
publishTelemetry is simply invoking async command via cmdlet and also not waiting for it(This is the way we have been using telemetry publish command). The only possible source of error can be the assert on agent version.